core/transcoder.go

```diff
@@ -45,6 +47,12 @@ func (lt *LocalTranscoder) Transcode(ctx context.Context, md *SegTranscodingMeta
 	// Returns UnrecoverableError instead of panicking to gracefully notify orchestrator about transcoder's failure
 	defer recoverFromPanic(&retErr)
 
+	// Validate input duration
+	// TODO: Allow 0 duration for backward compatibility; validate in the future.
+	if md.Duration.Seconds() > maxAllowedSegmentDuration.Seconds() || md.Duration.Seconds() < 0 {
```

Comment:
@leszko, @stronk-dev I now allow zero durations to prevent backward compatibility issues we can look into whether a zero duration also causes a problem later.